src/Particle/Par_Aux_Check_Particle.cpp

```diff
@@ -36,13 +58,16 @@ void Par_Aux_Check_Particle( const char *comment )
    long    ParID, NPar_Active_AllRank_Expect;
    double *EdgeL, *EdgeR;
    int     PassCheck[NCheck];
+   int    *Par_UID = new int [amr->Par->NextUID];
```

Comment:
In my opinion, the name `Par_UID` is a little confusing. It is not an array of UID, but it is an array of the counted number of particles for each UID. Would you think adding `Num` into the variable name would be more descriptive?